scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -856,14 +874,24 @@ object Execution {
    * these executions are executed in parallel: run is called on all at the
    * same time, not one after the other.
    */
+  private case class SequencingFn[T]() extends Function1[(T, List[T]), List[T]] {
```

Comment:
If we leave the anonymous closure, we won't get equality/etc.  So even for a fixed sequence, we won't have `Execution.sequence(a) == Execution.sequence(a)`.  As I'm sure you'd imagine, the same principle is behind `ReversingFn` and `nil`.

This does have the property that all instances will be equal (e.g., `SequencingFn[String]() == SequencingFn[Long]()`) but I don't think that's necessarily a problem since they really all do perform the same operation.

I don't feel very strongly about it, but I wanted to at least illustrate what would be necessary for that equality to work, since we could do it without changing the runtime semantics.